diff --git a/typo3/sysext/backend/Classes/Controller/FormInlineAjaxController.php b/typo3/sysext/backend/Classes/Controller/FormInlineAjaxController.php
index 83e239f..c137c58 100644
--- a/typo3/sysext/backend/Classes/Controller/FormInlineAjaxController.php
+++ b/typo3/sysext/backend/Classes/Controller/FormInlineAjaxController.php
@@ -105,6 +105,13 @@ class FormInlineAjaxController
             $childVanillaUid = (int)$inlineFirstPid;
         }
 
+        if (isset($ajaxArguments['context'])) {
+            $context = json_decode($ajaxArguments['context'], true);
+            if (GeneralUtility::hmac(serialize($context['config'])) === $context['hmac']) {
+                $parentConfig = $context['config'];
+            }
+        }
+        
         if ($parentConfig['type'] === 'flex') {
             $parentConfig = $this->getParentConfigFromFlexForm($parentConfig, $domObjectId);
         }
@@ -266,6 +273,13 @@ class FormInlineAjaxController
         $parentData = $formDataCompiler->compile($formDataCompilerInputForParent);
         $parentConfig = $parentData['processedTca']['columns'][$parentFieldName]['config'];
 
+        if (isset($ajaxArguments['context'])) {
+            $context = json_decode($ajaxArguments['context'], true);
+            if (GeneralUtility::hmac(serialize($context['config'])) === $context['hmac']) {
+                $parentConfig = $context['config'];
+            }
+        }
+        
         if ($parentConfig['type'] === 'flex') {
             $parentConfig = $this->getParentConfigFromFlexForm($parentConfig, $domObjectId, false);
             $parentData['processedTca']['columns'][$parentFieldName]['config'] = $parentConfig;
